boa_engine/src/value/mod.rs

```diff
@@ -1000,6 +1000,13 @@ impl From<f32> for Numeric {
     }
 }
 
+impl From<i128> for Numeric {
+    #[inline]
+    fn from(value: i128) -> Self {
+        Self::BigInt(value.into())
+    }
+}
+
```

Comment:
This should be a number (`f64`) instead of BigInt

Reply:
Well, we are already doing this with `i64`, and `BigInt` would make sure that it doesn't lose precision:
https://github.com/boa-dev/boa/blob/e2630faf82a27e958d0ca6c9b1c81554bcc07ff3/boa_engine/src/value/mod.rs#L1003-L1008

Reply:
> Well, we are already doing this with i64

Ummm.... This is wrong, It should be a number, not bigint.

> BigInt would make sure that it doesn't lose precision

It's fine if we lose precision (we lose precision if we type a number value which is too big for `f64` in JavaScript, same with `From<i64>` for `JsValue`), If we don't want loss of precision we create a bigint and we can convert it into `Numeric`.
The behaviour of `From<u/iN>` should be uniform, to do some number and others bigint is confuzing.

Reply:
There was some discussion here too: #1961

From my point of view, we should be able to convert an `f64` to `JsValue` or `Numeric`, and a `BigInt` too. But nothing else (at most an i32). If the user wants to convert a `i128` to a `BigInt`, they can first convert it to a `BigInt` and then convert the `BigInt` to a `Numeric` or a `JsValue`.

We shouldn't probably force anything, but give some options. Maybe a `TryFrom` implementation could be nice.

Reply:
> [](https://github.com/boa-dev/boa/issues/1961)
> 
> From my point of view, we should be able to convert an `f64` to `JsValue` or `Numeric`, and a `BigInt` too. But nothing else (at most an i32). If the user wants to convert a `i128` to a `BigInt`, they can first convert it to a `BigInt` and then convert the `BigInt` to a `Numeric` or a `JsValue`.
> 
> We shouldn't probably force anything, but give some options. Maybe a `TryFrom` implementation could be nice.

Yup. agreed